### PR TITLE
[CPDP-1440, CPDP-1448] - Default and Agent landing pages

### PR DESCRIPTION
**Pull request by modivilioglu**

### CODE REVIEW

File: conf/messages [hunk 1]
```diff
@@ -53,27 +53,127 @@ landingPage.createAccount.bullet2=update your contact details in your Capital Ga
...
+agentsLandingPage.contents.list1.ordered1=<a href="{0}">Who needs to report Capital Gains Tax on UK property</a>
+agentsLandingPage.contents.list1.ordered2=<a href="{0}">How to ask for authorisation</a>
+agentsLandingPage.contents.list1.ordered3=<a href="{0}">How to manage your client’s account</a>
+agentsLandingPage.contents.list1.ordered4=<a href="{0}">Send a Capital Gains Tax on UK property return</a>
+agentsLandingPage.contents.list1.ordered5=<a href="{0}">What you’ll need to complete the return</a>
+agentsLandingPage.contents.list1.ordered6=<a href="{0}">Ways your client can pay</a>
+agentsLandingPage.report.h3=Who needs to report Capital Gains Tax on UK property
+agentsLandingPage.report.uk-residents.h4=UK residents
+agentsLandingPage.report.uk-residents.p1=For UK residential property disposals made from 6 April 2020, UK residents have 30 days after the property's completion date to report and pay Capital Gains Tax on the disposals.
+agentsLandingPage.report.uk-residents.p2=<a href={0}>Check if your client needs to report and pay Capital Gains Tax on UK property</a>
```
Comment (AndyHWChung) [inline conf/messages:109]: missing quotes around for this href

File: conf/application.conf [hunk 2]
```diff
@@ -200,6 +200,21 @@ external-url {
...
   report-accessibility-problem="https://www.tax.service.gov.uk/contact/accessibility"
+  government-gateway-user-id-and-password="https://www.gov.uk/log-in-register-hmrc-online-services"
+  contact-hmrc="https://www.gov.uk/government/organisations/hm-revenue-customs/contact/capital-gains-tax-enquiries-for-individuals-employees-and-self-employed"
+  report-client-cgt-on-uk-property-disposals="https://www.tax.service.gov.uk/capital-gains-tax-uk-property/manage-clients-capital-gains-tax-on-uk-property-account"
+  trust-registered="https://www.gov.uk/trusts-taxes/trustees-tax-responsibilities"
+  non-resident-report-and-pay-cgt-on-uk-property-or-land-form="https://www.tax.service.gov.uk/shortforms/form/NRCGT_Return"
+  personal-representative="https://www.gov.uk/government/publications/death-personal-representatives-and-legatees-hs282-self-assessment-helpsheet/hs282-death-personal-representatives-and-legatees-2018"
+  who-needs-to-report-cgt-on-uk-property="https://www.tax.service.gov.uk/capital-gains-tax-uk-property/manage-clients-capital-gains-tax-on-uk-property-account#who-needs-to-report-capital-gains-tax-on-uk-property"
```
[the remark this answers is not in the document]
Comment (AndyHWChung) [reply conf/application.conf:209]: (you can give the `h2` the id's to link to)

File: conf/app.routes [hunk 3]
```diff
@@ -1,5 +1,6 @@
 GET /create-account/need-more-details                                       uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.StartController.weNeedMoreDetails()
 GET /start/report-pay-capital-gains-tax-uk-property                         uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.LandingPageController.landingPage()
+GET /start/report-pay-capital-gains-tax-uk-property/agents                  uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.LandingPageController.agentsLandingPage()
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply conf/app.routes:3]: Done.

File: conf/application.conf [hunk 4]
```diff
@@ -200,6 +200,21 @@ external-url {
   equality-commission-for-northen-ireland="https://www.equalityni.org/Home"
   digital-accessibility-centre="http://www.digitalaccessibilitycentre.org/"
   report-accessibility-problem="https://www.tax.service.gov.uk/contact/accessibility"
+  government-gateway-user-id-and-password="https://www.gov.uk/log-in-register-hmrc-online-services"
+  contact-hmrc="https://www.gov.uk/government/organisations/hm-revenue-customs/contact/capital-gains-tax-enquiries-for-individuals-employees-and-self-employed"
+  report-client-cgt-on-uk-property-disposals="https://www.tax.service.gov.uk/capital-gains-tax-uk-property/manage-clients-capital-gains-tax-on-uk-property-account"
+  trust-registered="https://www.gov.uk/trusts-taxes/trustees-tax-responsibilities"
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply conf/application.conf:206]: Done.

File: conf/application.conf [hunk 5]
```diff
@@ -200,6 +200,21 @@ external-url {
...
+  non-resident-report-and-pay-cgt-on-uk-property-or-land-form="https://www.tax.service.gov.uk/shortforms/form/NRCGT_Return"
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply conf/application.conf:207]: Done.

File: conf/messages (hunk 1 above)
Comment (modivilioglu) [reply conf/messages:109]: Done.

File: conf/application.conf (hunk 2 above)
[the remark this answers is not in the document]
Comment (modivilioglu) [reply conf/application.conf]: Done. (h3)

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/agents_landing_page.scala.html [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    <ol class="list list-number">
+        <li>@Html(messages("agentsLandingPage.contents.list1.ordered1", s"#${appConfig.whoNeedsToReportCgtOnUkProperty}"))</li>
+        <li>@Html(messages("agentsLandingPage.contents.list1.ordered2", s"#${appConfig.howToAskForAuthorisation}" ))</li>
+        <li>@Html(messages("agentsLandingPage.contents.list1.ordered3", s"#${appConfig.howToManageYourClientsAccount}"))</li>
+        <li>@Html(messages("agentsLandingPage.contents.list1.ordered4", s"#${appConfig.sendACgtOnUkPropertyReturn}"))</li>
+        <li>@Html(messages("agentsLandingPage.contents.list1.ordered5", s"#${appConfig.whatYouWillNeedToCompleteTheReturn}"))</li>
+        <li>@Html(messages("agentsLandingPage.contents.list1.ordered6", s"#${appConfig.waysYourClientCanPay}"))</li>
+    </ol>
+
+    <h3 id="@{appConfig.whoNeedsToReportCgtOnUkProperty}" class="heading-medium">@messages("agentsLandingPage.report.h3")</h3>
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/agents_landing_page.scala.html:65]: you don't need to read the id from config - just hard code it here

File: conf/application.conf [hunk 7]
```diff
@@ -200,6 +200,20 @@ external-url {
   equality-commission-for-northen-ireland="https://www.equalityni.org/Home"
   digital-accessibility-centre="http://www.digitalaccessibilitycentre.org/"
   report-accessibility-problem="https://www.tax.service.gov.uk/contact/accessibility"
+  government-gateway-user-id-and-password="https://www.gov.uk/log-in-register-hmrc-online-services"
+  contact-hmrc="https://www.gov.uk/government/organisations/hm-revenue-customs/contact/capital-gains-tax-enquiries-for-individuals-employees-and-self-employed"
+  report-client-cgt-on-uk-property-disposals="https://www.tax.service.gov.uk/capital-gains-tax-uk-property/manage-clients-capital-gains-tax-on-uk-property-account"
```
Comment (AndyHWChung) [inline conf/application.conf:205]: this is one of our urls. If the url has `www.tax.service.gov.uk/capital-gains-tax-uk-property` in it that means it's ours. We shouldn't hard code our own url's - use the reverse routes to give you a url instead.

File: conf/application.conf [hunk 8]
```diff
@@ -200,6 +200,20 @@ external-url {
...
+  trust-registered="https://www.gov.uk/trusts-taxes/trustees-tax-responsibilities"
+  personal-representative="https://www.gov.uk/government/publications/death-personal-representatives-and-legatees-hs282-self-assessment-helpsheet/hs282-death-personal-representatives-and-legatees-2018"
+  who-needs-to-report-cgt-on-uk-property="who-needs-to-report-capital-gains-tax-on-uk-property"
```
Comment (AndyHWChung) [inline conf/application.conf:208]: you don't need these id's in config

File: conf/messages [hunk 9]
```diff
@@ -53,27 +53,127 @@ landingPage.createAccount.bullet2=update your contact details in your Capital Ga
...
+agentsLandingPage.contents.h4=Contents
+agentsLandingPage.contents.list1.ordered1=<a href="{0}">Who needs to report Capital Gains Tax on UK property</a>
+agentsLandingPage.contents.list1.ordered2=<a href="{0}">How to ask for authorisation</a>
+agentsLandingPage.contents.list1.ordered3=<a href="{0}">How to manage your client’s account</a>
+agentsLandingPage.contents.list1.ordered4=<a href="{0}">Send a Capital Gains Tax on UK property return</a>
+agentsLandingPage.contents.list1.ordered5=<a href="{0}">What you’ll need to complete the return</a>
+agentsLandingPage.contents.list1.ordered6=<a href="{0}">Ways your client can pay</a>
+agentsLandingPage.report.h3=Who needs to report Capital Gains Tax on UK property
+agentsLandingPage.report.uk-residents.h4=UK residents
+agentsLandingPage.report.uk-residents.p1=For UK residential property disposals made from 6 April 2020, UK residents have 30 days after the property's completion date to report and pay Capital Gains Tax on the disposals.
```
Comment (AndyHWChung) [inline conf/messages:108]: is that a straight quote in `property's`?

File: conf/messages [hunk 10]
```diff
@@ -53,27 +53,127 @@ landingPage.createAccount.bullet2=update your contact details in your Capital Ga
...
+agentsLandingPage.report.uk-residents.p2=<a href="{0}">Check if your client needs to report and pay Capital Gains Tax on UK property</a>
+agentsLandingPage.report.uk-residents.p3=For UK residential property disposals made from 6 April 2020, UK residents must report their Capital Gains Tax online. You can do this for your clients through their Capital Gains Tax on UK property account.
+agentsLandingPage.report.uk-residents.p4=If your client is a resident who is an individual or a trust, you can use this service to:
+
+agentsLandingPage.report.uk-residents.list1.bullet1=report your client’s UK residential property disposals made from 6 April 2020
+agentsLandingPage.report.uk-residents.list1.bullet2=tell your client how to pay any tax they might owe for that disposal
+agentsLandingPage.report.uk-residents.list1.bullet3=view and amend your client’s previous and current returns
+agentsLandingPage.report.what-residents-cannot-do.h4=What residents cannot do
+agentsLandingPage.report.what-residents-cannot-do.p1=You cannot use the online service for your client to report UK property disposals for someone who has died - you’ll need to <a href="{0}">contact HMRC.</a>
```
Comment (AndyHWChung) [inline conf/messages:117]: the full stop should sit outside of the link

File: conf/messages [hunk 11]
```diff
@@ -53,27 +53,127 @@ landingPage.createAccount.bullet2=update your contact details in your Capital Ga
...
+agentsLandingPage.how-to-ask-for-authorisation.p3=Sign in to your agent services account then select ‘ask a client to authorise you’.
+agentsLandingPage.how-to-ask-for-authorisation.p4=Enter your client’s details. You'll then get a link to share with your client. Send your client an email asking them to go to the link to authorise you. The link will expire 14 days after it was created.
+agentsLandingPage.how-to-ask-for-authorisation.p5=Tell your client they’ll need the Government Gateway user ID and password they use for their personal tax affairs (if they’re an individual) or the trust name (if they’re a trust). If they do not have a user ID, they can create one when they go to your link.
+agentsLandingPage.how-to-ask-for-authorisation.p6=When your client selects the link in the email you sent them, you’ll then be able to manage their Capital Gains Tax on UK property account and returns.
+agentsLandingPage.how-to-ask-for-authorisation.p7=You'll get an email after your client accepts or rejects your request, or if the link has expired after 14 days.
```
Comment (AndyHWChung) [inline conf/messages:139]: is that a straight quote in `You'll`?

File: conf/messages [hunk 12]
```diff
@@ -53,27 +53,127 @@ landingPage.createAccount.bullet2=update your contact details in your Capital Ga
...
+agentsLandingPage.send-a-cgt-on-uk-property-return.p2=The return will be saved in your client’s account home for 30 days from the last day it was saved.
+agentsLandingPage.send-a-cgt-on-uk-property-return.p3=If your client disposed of more than one property from 6 April 2020 with the same completion date, you can use this service to report the Capital Gains Tax for all those properties. The properties must also have exchange dates in the same tax year.
+
+agentsLandingPage.what-you-will-need-to-complete-the-return.h3=What you’ll need to complete the return
+agentsLandingPage.what-you-will-need-to-complete-the-return.p1=You need these details about your client’s disposal:
+agentsLandingPage.what-you-will-need-to-complete-the-return.list1.bullet1=address and postcode
+agentsLandingPage.what-you-will-need-to-complete-the-return.list1.bullet2=date they got the property
+agentsLandingPage.what-you-will-need-to-complete-the-return.list1.bullet3=date they exchanged contracts when selling or disposing of the property
+agentsLandingPage.what-you-will-need-to-complete-the-return.list1.bullet4=date they stopped being the property's owner (completion date)
```
Comment (AndyHWChung) [inline conf/messages:165]: looks like a straight quote again

File: conf/application.conf (hunk 7 above)
Comment (modivilioglu) [reply conf/application.conf]: Done.

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**